lib/compiler/src/beam_validator.erl

```diff
@@ -995,6 +995,18 @@ vi(raw_raise=I, Vst0) ->
 
 vi(bs_init_writable=I, Vst) ->
     validate_body_call(I, 1, Vst);
+vi({bs_create_bin,{f,Fail},Heap,Live,Unit,Dst,{list,List}}, Vst0) ->
+    verify_live(Live, Vst0),
+    verify_y_init(Vst0),
+    verify_create_bin_list(List, Vst0),
+    Vst = heap_alloc(Heap, Vst0),
```

Comment:
Does the allocation stick on both failure and success? If not, this ought to be moved to the success branch, and if we can fail both after and before allocating, the failure branch ought to have `heap_alloc(0, FailVst)` to ensure that we don't leave anything there.

I noticed that the old instructions lack this distinction too. It hasn't caused any trouble as far as I can tell, but it struck me as slightly off.